core/src/runtime/opaque.rs

```diff
@@ -37,24 +51,63 @@ impl<'js> Opaque<'js> {
     pub fn new() -> Self {
         Opaque {
             panic: Cell::new(None),
+
             interrupt_handler: UnsafeCell::new(None),
+
+            class_id: qjs::JS_INVALID_CLASS_ID,
+            callable_class_id: qjs::JS_INVALID_CLASS_ID,
+
+            prototypes: UnsafeCell::new(HashMap::new()),
+
             userdata: UserDataMap::default(),
+
+            _marker: PhantomData,
+
             #[cfg(feature = "futures")]
             spawner: None,
-            _marker: PhantomData,
         }
     }
 
     #[cfg(feature = "futures")]
     pub fn with_spawner() -> Self {
-        Opaque {
-            panic: Cell::new(None),
-            interrupt_handler: UnsafeCell::new(None),
-            userdata: UserDataMap::default(),
-            #[cfg(feature = "futures")]
-            spawner: Some(UnsafeCell::new(Spawner::new())),
-            _marker: PhantomData,
+        let mut this = Opaque::new();
+        this.spawner = Some(UnsafeCell::new(Spawner::new()));
+        this
+    }
+
+    pub unsafe fn initialize(&mut self, rt: *mut qjs::JSRuntime) -> Result<(), Error> {
+        qjs::JS_NewClassID((&mut self.class_id) as *mut qjs::JSClassID);
+        qjs::JS_NewClassID((&mut self.callable_class_id) as *mut qjs::JSClassID);
+
+        let class_def = qjs::JSClassDef {
+            class_name: b"RustClass\0".as_ptr().cast(),
```

Comment:
Should we also use a more private name, i.e `__qjs_rc` and put this as a const some where?

Reply:
We can, but it doesn't really matter. `RustClass` is not accessable anywhere, it is not stored in the global object or anywhere else so it can't pollute a namespace. 
It also can't cause conflicts as you can register two classes with the same class_name without problems. 

The only place you will see this name is in dumps when enabling the debug dump features. It might be a good idea to make this a bit more descriptive, maybe `RquickjsClass` to clearly signify where the class came from.